homeassistant/components/reolink/util.py

```diff
@@ -0,0 +1,23 @@
+"""Utility functions for the Reolink component."""
+from __future__ import annotations
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+
+from . import ReolinkData
+from .const import DOMAIN
+
+
+def has_connection_problem(
+    hass: HomeAssistant, config_entry: config_entries.ConfigEntry
+) -> bool:
+    """Check if a existing entry has a connection problem."""
+    reolink_data: ReolinkData | None = hass.data.get(DOMAIN, {}).get(
+        config_entry.entry_id
+    )
+    connection_problem = (
+        reolink_data is not None
+        and config_entry.state == config_entries.ConfigEntryState.LOADED
+        and reolink_data.device_coordinator.last_update_success
```

Comment:
The resulting boolean doesn't match the name of this function.

Reply:
Does 'has_update_error' sound better, or do you have another suggestion for the name of the function?
Or do you mean I schould rename the boolean to `has_connection_problem`?

Reply:
The meaning is inverted compared to the resulting boolean.

Reply:
oh whoops, just see it now.
Will fix it.
At least the logic itself is good since the name is also used wrong in the config_flow

Reply:
Fix is here: https://github.com/home-assistant/core/pull/100113